components/chart/editModule/index.tsx

```diff
@@ -0,0 +1,282 @@
+  const drawerRef = useRef(null);
+  const editPanelRef = useRef(null);
+  const [successMessage, setSuccessMessage] = useState<string>('');
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const { data, loading } = useContentInstancesQuery(module?.id);
+  const { data: modelsResult, loading: modelsLoading } =
+    useModelsQuery(TEMPLATE_CONTEXT_ID);
+  const [
+    updateContentInstanceFieldValues,
+    {
+      data: updateContentInstanceFieldValuesCallData,
+      error: updateContentInstanceFieldValuesCallError,
+      loading: updateContentInstanceFieldValuesLoading,
+    },
+  ] = useUpdateContentInstanceFieldValuesMutation();
+
+  const targetModel = useMemo(() => {
+    if (modelsResult?.viewTypes) {
+      return modelsResult.viewTypes.find(viewType => viewType.Id === HYCHART_VIEW_TYPE_ID);
+    }
+    return undefined;
+  }, [modelsResult, module?.id]);
+
+  const { control, handleSubmit, formState: { errors }, register, setValue, getValues } = useForm<any>();
+  useEffect(() => {
+    if (targetModel && data?.contentInstances.length > 0) {
+      targetModel?.ViewFields.reduce(
+        (obj, field) => {
+          const fieldData = data.contentInstances[0].FieldsValues.find(
+            (f) => (f.FieldId === field.Id),
+          );
+          if (field.Type === ModelFieldsTypes.PredefinedListCheckboxes) {
+            const fieldDataForCheckboex = data.contentInstances[0].FieldsValues.filter(
+              (f) => f.FieldId === field.Id,
+            );
+            const finishedCheckedBox = fieldDataForCheckboex.map((checkBoxValue) => checkBoxValue.Value.split(','))
+            setValue(`${field.Id}`, finishedCheckedBox.flat() || '')
+          } else if (field.Type === ModelFieldsTypes.Boolean) {
+            data.contentInstances[0].FieldsValues.filter(
+              (f) => f.FieldId === field.Id,
+            ).map((item) => {
+              setValue(`${field.Id}`, item.Value === 'True' ? true : false)
+            });
+          } else {
+            setValue(`${field.Id}`, fieldData?.Value || '')
+          }
+
+          return {
+            ...obj,
+            [`${field.Id}`]: fieldData?.Value || '',
+          };
+        },
+        {}
+      );
+    }
+  }, [data, targetModel]);
+
+  const onSubmit = (formData: FieldValues) => {
+    const updatedFormData: FieldValues = { ...formData };
+    let ci: number;
+    if (targetModel && data?.contentInstances.length > 0) {
+      targetModel?.ViewFields.reduce(
+        (obj, field) => {
+          if (field.Type === ModelFieldsTypes.PredefinedListCheckboxes) {
+            updatedFormData[field.Id] = formData[field.Id].join(',')
+          } else if (field.Type === ModelFieldsTypes.Boolean) {
+            updatedFormData[field.Id] = formData[field.Id] === true ? 'True' : 'False'
+          }
+          data.contentInstances[0].FieldsValues.map((details, index) => {
+            ci = details.ContentId;
+          });
```

Comment:
I think no need for this map, it's running many times to save the content instance ID again and again, I suggest here have something like the next
```suggestion
const targetContentInstance = useMemo(() => {
    if (data?.contentInstances.length > 0) {
      return data.contentInstances[0];
    }
    return undefined;
  }, [data]);
```